migrations/1601022564938-copy-parents-data-into-children-entities-and-delete-parent-users.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-await-in-loop */
+// eslint-disable-next-line no-unused-vars
+const mongoose = require('mongoose');
+
+const { info } = require('../src/logger');
+
+const { connect, close } = require('../src/utils/database');
+
+const parentSchema = new mongoose.Schema({
+	firstName: { type: String, required: true },
+	lastName: { type: String, required: true },
+	email: { type: String, required: true, lowercase: true },
+});
+
+const consentForm = ['analog', 'digital', 'update'];
+
+const OldUser = mongoose.model(
+	'oldUser',
+	new mongoose.Schema({
+		parents: [{ type: mongoose.Schema.Types.ObjectId, ref: 'user' }],
+		consent: {
+			parentConsents: [
+				{
+					parentId: { type: mongoose.Schema.Types.ObjectId, ref: 'user' },
+					form: { type: String, enum: consentForm },
+					dateOfPrivacyConsent: { type: Date },
+					dateOfTermsOfUseConsent: { type: Date },
+					privacyConsent: { type: Boolean },
+					termsOfUseConsent: { type: Boolean },
+				},
+			],
+		},
+	}),
+	'users'
+);
+
+const User = mongoose.model(
+	'myUserModel',
+	new mongoose.Schema({
+		roles: [{ type: mongoose.Schema.Types.ObjectId, ref: 'role' }],
+		parents: [parentSchema],
+		consent: {
+			parentConsents: [
+				{
+					parentId: { type: mongoose.Schema.Types.ObjectId, ref: 'user' },
+					form: { type: String, enum: consentForm },
+					dateOfPrivacyConsent: { type: Date },
+					dateOfTermsOfUseConsent: { type: Date },
+					privacyConsent: { type: Boolean },
+					termsOfUseConsent: { type: Boolean },
+				},
+			],
+		},
+	}),
+	'users'
+);
+
+const getIds = (doc) =>
+	doc.reduce((prev, curr) => {
```

Comment:
Why not just a simple 'map' function instead of reduce?

Reply:
done